bhyve.pyx

```diff
@@ -1,4 +1,92 @@
 cimport bhyve
+cimport types
+cimport vmm
 
+import enum
 
+from libc.errno cimport errno
+from libc.stdlib cimport free
 
+
+class Error(enum.IntEnum):
+    OPEN_FAILED = 1
+    STATS_FETCH_FAILED = 2
+
+
+class BhyveException(RuntimeError):
+    def __init__(self, code, message):
+        super(BhyveException, self).__init__(f'[Errorno {code}] {message}')
+        self.code = code
+
+cdef class VM:
+
+    cdef str vm_name
+    cdef bhyve.vmctx* vm
+
+    def __cinit__(self, str vm_name, bint create_vm=0):
+        self.vm_name = vm_name
+
+        if create_vm:
+            if bhyve.vm_create(vm_name.encode()) != 0:
```

Comment:
Calling C functions (.e.g bhyve.*) should release the GIL (with nogil:). So these can run in a thread and not block other python code to run.